python/CMakeLists.txt

```diff
@@ -1,9 +1,12 @@
 find_package(pybind11 2.2 CONFIG)
 
 if(NOT pybind11_FOUND)
-  add_subdirectory( pybind11 )
-  # SYSTEM is supported for embedded pybind source (>= 2.3.0)
-  set(PYBIND11_SYSTEM "SYSTEM")
+  include(FetchContent)
+  FetchContent_Declare(pybind11
+                       URL https://github.com/pybind/pybind11/archive/refs/tags/v2.11.0.tar.gz
+                       URL_HASH SHA512=10edfdc499ed70234de0796f79b9f3a7a14d4baf01fb09fedbd9126e0a13d9cdeb213950c53a8f7a67cd6049b1db31efacfe2192098236b242ad3685967ea907
+                       DOWNLOAD_EXTRACT_TIMESTAMP ON)
```

Comment:
Same comment as for https://github.com/OPM/opm-common/pull/3671, it works if   `DOWNLOAD_EXTRACT_TIMESTAMP ON` is put before `URL` and `URL_HASH` properties (tested on Ubuntu 22.04, cmake version 3.22.1)